src/libs/ReportUtils.js

```diff
@@ -1475,28 +1475,75 @@ function getTransactionDetails(transaction, createdDateFormat = CONST.DATE.FNS_F
  * Can only edit if:
  *
  * - in case of IOU report
- *    - the current user is the requestor
+ *    - the current user is the requestor and is not settled yet
  * - in case of expense report
- *    - the current user is the requestor
+ *    - the current user is the requestor and is not settled yet
  *    - or the user is an admin on the policy the expense report is tied to
  *
  * @param {Object} reportAction
  * @returns {Boolean}
  */
 function canEditMoneyRequest(reportAction) {
-    // If the report action i snot IOU type, return true early
+    const isDeleted = ReportActionsUtils.isDeletedAction(reportAction);
+
+    if (isDeleted) {
+        return false;
+    }
+
+    // If the report action is not IOU type, return true early
     if (reportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
         return true;
     }
+
     const moneyRequestReportID = lodashGet(reportAction, 'originalMessage.IOUReportID', 0);
+
     if (!moneyRequestReportID) {
         return false;
     }
+
     const moneyRequestReport = getReport(moneyRequestReportID);
     const isReportSettled = isSettled(moneyRequestReport.reportID);
     const isAdmin = isExpenseReport(moneyRequestReport) && lodashGet(getPolicy(moneyRequestReport.policyID), 'role', '') === CONST.POLICY.ROLE.ADMIN;
     const isRequestor = currentUserAccountID === reportAction.actorAccountID;
-    return !isReportSettled && (isAdmin || isRequestor);
+
+    if (isAdmin) {
+        return true;
+    }
+
+    return !isReportSettled && isRequestor;
+}
+
+/**
+ * Checks if the current user can edit the provided property of a money request
+ *
+ * @param {Object} reportAction
+ * @param {String} reportID
+ * @param {String} fieldToEdit
+ * @returns {Boolean}
+ */
+function canEditFieldOfMoneyRequest(reportAction, reportID, fieldToEdit) {
+    // A list of fields that cannot be edited by anyone, once a money request has been settled
+    const nonEditableFieldsWhenSettled = [
+        CONST.EDIT_REQUEST_FIELD.AMOUNT,
+        CONST.EDIT_REQUEST_FIELD.CURRENCY,
+        CONST.EDIT_REQUEST_FIELD.DATE,
+        CONST.EDIT_REQUEST_FIELD.RECEIPT,
+        CONST.EDIT_REQUEST_FIELD.DISTANCE,
+    ];
+
+    // Checks if this user has permissions to edit this money request
+    if (!canEditMoneyRequest(reportAction)) {
+        return false; // User doesn't have permission to edit
+    }
+
+    // Checks if the report is settled
+    // Checks if the provided property is a restricted one
+    if (!isSettled(reportID) || !nonEditableFieldsWhenSettled.includes(fieldToEdit)) {
+        return true; // User can edit
+    }
+
+    // Current user cannot edit the provided property of the money request
+    return false;
```

Comment:
NAB: For this last case, I think you can do it more simply still:
```suggestion
    // Checks if the report is settled
    // Checks if the provided property is a restricted one
    return !isSettled(reportID) || !nonEditableFieldsWhenSettled.includes(fieldToEdit);
```

Reply:
Done - 8f48a90.